infrastructure/memory/sqlite3.go

```
@@ -205,3 +205,64 @@ func (e *statementExecutor) ExecuteStatement(ctx context.Context, sql *model.SQL
+
+// GetTableDDL get table DDL in memory
+func (d *ddlGetter) GetTableDDL(ctx context.Context, tableName string) ([]*model.Table, error) {
+	tx, err := d.db.BeginTx(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+	defer tx.Rollback()
+
+	// PRAGMA table_info returns: cid, name, type, notnull, dflt_value, pk
+	query := "PRAGMA table_info(" + tableName + ")"
+	rows, err := tx.QueryContext(ctx, query)
```

golangci-lint (1.64.8)

rows.Err must be checked

(rowserrcheck)